src/App/Exercises/Js/EmptyValuesAndComments/EmptyValuesAndComments.jsx

```diff
@@ -0,0 +1,86 @@
+import { useState } from 'react';
+import OutlinedInput from '@mui/material/OutlinedInput';
+
+import './styles.css';
+export const EmptyValuesAndComments = () => {
+  // const myText = '  Ala ma kota   ';
+  // const myArray = [];
+  // const myTextEdit = myText.trim();
+
+  // console.log(myText.toUpperCase());
+  // console.log(myText.toLowerCase());
+  // console.log(myText.length);
+  // console.log(myText.trim()); // usuwa biale znaki
+  // console.log(myText.indexOf('a'));
+  // console.log(myTextEdit.replace(' ', '_'));
+  // console.log(myTextEdit.replaceAll(' ', '_'));
+  // console.log(myText.codePointAt(0));
+
+  // console.log(myArray.length);
+
+  let myNewText = 'ala_ma_kota_a_tomek_ma_psa';
+  const [text, setText] = useState(myNewText);
+  const [pixels, setPixels] = useState('');
+  const [height, setHeight] = useState('');
+  const [width, setWidth] = useState('');
+  const [flag, setFlag] = useState(0);
+  // const [typOf, setTypeOf] = useState('');
+
+  function myTextReplace() {
+    myNewText = myNewText.replaceAll('_', ' ');
+  }
+  function addChartText() {
+    myNewText = myNewText + '.';
+  }
+  function firstLetterUpperCase() {
```

Comment:
Fajnie, że idziesz w podejście, że każda funkcja robi jedno i tylko jedno 😃 tylko wyobraź sobie teraz, że np. tą funkcję miałbyś użyć gdzieś indziej. No nie użyjesz, bo ona robi zmiany tylko na zmiennej `myNewText`. Wg mnie lepiej byłoby, żeby takie funkcje przyjmowały text w parametrze i zwracały zmienioną wartość 😃 

Ale patrząc na to, że mamy te funkcje wewnątrz componentu i jeżeli wiesz, że nigdzie indziej tego nie użyjesz, to może to tak zostać, bo działa 😄

Reply:
Ok, dzięki za uwagi. Przerobię żeby wyrabiać w sobie przyzwyczajenie ; )